.github/workflows/check-upstream-upgrade.yml

```diff
@@ -0,0 +1,40 @@
+# WARNING: This file was build based on https://github.com/pulumi/ci-mgmt
+# but it is currently not in sync. Any update must be done manually
+
+env:
+  GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+jobs:
+  check_upgrade_provider:
+    name: Check for upstream provider upgrades
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout Repo
+      uses: actions/checkout@v4
+      with:
+        submodules: true
+    - name: Setup tools
+      uses: ./.github/actions/setup-tools
+      with:
+        tools: go
+    - name: Install upgrade-provider
+      run: go install github.com/pulumi/upgrade-provider@main
+      shell: bash
+    - name: "Set up git identity: name"
+      run: git config --global user.name GitHub
+      shell: bash
+    - name: "Set up git identity: email"
+      run: git config --global user.email noreply@github.com
+      shell: bash
+    - name: Run upgrade-provider upstream check
+      id: upstream_version
+      run: |
+        upgrade-provider "$REPO" --kind=check-upstream-version
```

Comment:
Just to confirm: the `upgrade-provider` command runs all necessary make tasks & patches and then opens a GitHub PR with the resulting changes?

Reply:
exactly. This action executes it with kind=check-upstream-version, If a change is detected, it will create a new issue with the title "Upgrade terraform-provider-... version x.y.z". That will trigger the upgrade-provider action https://github.com/equinix/pulumi-equinix/blob/410b29091120c3e2109e8d2e454a11451b0eaae8/.github/workflows/upgrade-provider.yml#L15 which internally also uses the upgrade-provider command but with the `kind=all`, this updates both the terraform provider, the bridge plugin and its dependencies